lvgl/src/lib.rs

```diff
@@ -47,22 +44,60 @@ pub(crate) mod mem;
 #[cfg(not(feature = "lvgl_alloc"))]
 use crate::mem::Box;
 
+mod display;
+pub use display::*;
+mod functions;
+mod support;
+mod ui;
+pub mod widgets;
+use core::sync::atomic::{AtomicBool, Ordering};
+pub use functions::*;
 pub use lv_core::*;
 pub use support::*;
 pub use ui::*;
 
-use core::sync::atomic::{AtomicBool, Ordering};
+struct RunOnce(AtomicBool);
 
-// Initialize LVGL only once.
-static LVGL_INITIALIZED: AtomicBool = AtomicBool::new(false);
+impl RunOnce {
+    const fn new() -> Self {
+        Self(AtomicBool::new(false))
+    }
 
-pub(crate) fn lvgl_init() {
-    if LVGL_INITIALIZED
-        .compare_exchange(false, true, Ordering::Relaxed, Ordering::Relaxed)
-        .is_ok()
-    {
+    fn swap_and_check(&self) -> bool {
+        self.0
+            .compare_exchange(false, true, Ordering::Relaxed, Ordering::Relaxed)
+            .is_ok()
+    }
+}
+
+static LVGL_INITIALIZED: RunOnce = RunOnce::new();
+
+pub fn init() {
+    if LVGL_INITIALIZED.swap_and_check() {
         unsafe {
             lvgl_sys::lv_init();
         }
     }
 }
+
+#[cfg(test)]
+pub(crate) mod tests {
+    use super::*;
+    use crate::display::{Display, DrawBuffer};
+    use embedded_graphics::mock_display::MockDisplay;
+    use embedded_graphics::pixelcolor::Rgb565;
+
+    pub(crate) fn initialize_test() {
+        init();
```

Comment:
All tests run in the same LVGL "instance" so the tests might run out of memory at some point. :) Just to keep in mind.